tags/tag_set_builder.go

```diff
@@ -42,60 +34,45 @@ func NewTagSetBuilder(ts *TagSet) TagSetBuilder {
 
 	tb.ts = newTagSet(len(ts.m))
 	for k, b := range ts.m {
-		tb.ts.upsertBytes(k, b)
+		tb.ts.setKeyValue(k, b)
```

Comment:
IMO I would keep calling the methods on tagSet as they will be a lot more complex in the future and it is better to have them close to the tagSet instead of here. Look at my comment on removing the tagSet methods.
And you'll probably need to repeat that same complex logic in the decoders as there you don't need to go thru the builder

Reply:
Done, got them back.